doc/source/custom_directives.py

```diff
@@ -0,0 +1,217 @@
+# Originally from:
+# github.com/pytorch/tutorials/blob/60d6ef365e36f3ba82c2b61bf32cc40ac4e86c7b/custom_directives.py # noqa
+from docutils.parsers.rst import Directive, directives
+from docutils.statemachine import StringList
+from docutils import nodes
+import re
+import os
+import sphinx_gallery
+
+try:
+    FileNotFoundError
+except NameError:
+    FileNotFoundError = IOError
+
+
+class IncludeDirective(Directive):
+    """Include source file without docstring at the top of file.
+
+    Implementation just replaces the first docstring found in file
+    with '' once.
+
+    Example usage:
+
+    .. includenodoc:: /beginner/examples_tensor/two_layer_net_tensor.py
+
+    """
+
+    # defines the parameter the directive expects
+    # directives.unchanged means you get the raw value from RST
+    required_arguments = 1
+    optional_arguments = 0
+    final_argument_whitespace = True
+    has_content = False
+    add_index = False
+
+    docstring_pattern = r'"""(?P<docstring>(?:.|[\r\n])*?)"""\n'
+    docstring_regex = re.compile(docstring_pattern)
+
+    def run(self):
+        document = self.state.document
+        env = document.settings.env
+        rel_filename, filename = env.relfn2path(self.arguments[0])
+
+        try:
+            text = open(filename).read()
+            text_no_docstring = self.docstring_regex.sub('', text, count=1)
+
+            code_block = nodes.literal_block(text=text_no_docstring)
+            return [code_block]
+        except FileNotFoundError as e:
+            print(e)
+            return []
+
+
+class GalleryItemDirective(Directive):
```

Comment:
If you got this from somewhere can you link to it please?

Reply:
already linked in top line of file